File: Iteration_2/src/main/java/MandatoryCourse.java
public class MandatoryCourse extends Course {

    public MandatoryCourse(String courseId, String courseName, int credits, Course prerequisiteCourse, int semester) {
        super(courseId, courseName, credits, prerequisiteCourse, semester);
    }
    public MandatoryCourse(String courseId, String courseName, String grade, int credits) {
        super(courseId, courseName, grade, credits);
    }
    public MandatoryCourse(String courseId, String courseName, int credits) {
        super(courseId, courseName, credits);
    }

    @Override
    public String getCourseType() {
        return "Mandatory";
    }
}

File: Iteration_3/MandatoryCourse.py
from Course import Course

class MandatoryCourse(Course):
    def __init__(self, course_id: str, course_name: str, credits: int, prerequisite_course: 'Course' = None, semester: int = None):
        if prerequisite_course and semester is not None:
            super().__init__(course_id, course_name, credits, prerequisite_course, semester)
        elif semester is not None:
            super().__init__(course_id, course_name, credits, semester)
        elif prerequisite_course:
            super().__init__(course_id, course_name, credits, prerequisite_course)
        else:
            super().__init__(course_id, course_name, credits)

    def get_course_type(self):
        return "Mandatory"
